File: edukate-common/src/main/java/io/github/sanyavertolet/edukate/common/services/NoopNotifier.java
package io.github.sanyavertolet.edukate.common.services;

import io.github.sanyavertolet.edukate.common.notifications.BaseNotificationCreateRequest;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Slf4j
@Component
@Profile("!notifier")
@AllArgsConstructor
public class NoopNotifier implements Notifier {

    @Override
    public Mono<String> notify(BaseNotificationCreateRequest notificationCreationRequest) {
        log.warn("Notification is meant to be send: {}", notificationCreationRequest);
        return Mono.just("stub");
    }
}

File: edukate-common/src/main/kotlin/io/github/sanyavertolet/edukate/common/services/NoopNotifier.kt
package io.github.sanyavertolet.edukate.common.services

import io.github.sanyavertolet.edukate.common.notifications.BaseNotificationCreateRequest
import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component
import reactor.core.publisher.Mono
import reactor.kotlin.core.publisher.toMono

@Component
@Profile("!notifier")
class NoopNotifier : Notifier {
    override fun notify(notificationCreationRequest: BaseNotificationCreateRequest): Mono<String> =
        "stub".toMono().also { logger.warn("Notification is meant to be send: {}", notificationCreationRequest) }

    companion object {
        private val logger = LoggerFactory.getLogger(NoopNotifier::class.java)
    }
}
